test/integration/hds_integration_test.cc

```diff
@@ -548,15 +574,52 @@ TEST_P(HdsIntegrationTest, TwoEndpointsDifferentLocality) {
 
   // Receive updates until the one we expect arrives
   ASSERT_TRUE(hds_stream_->waitForGrpcMessage(*dispatcher_, response_));
-  while (!(checkEndpointHealthResponse(response_.endpoint_health_response().endpoints_health(0),
+  while (!(checkEndpointHealthResponse(response_.endpoint_health_response()
+                                           .cluster_endpoints_health(0)
+                                           .locality_endpoints_health(0)
+                                           .endpoints_health(0),
                                        envoy::config::core::v3::UNHEALTHY,
                                        host_upstream_->localAddress()) &&
-           checkEndpointHealthResponse(response_.endpoint_health_response().endpoints_health(1),
+           checkEndpointHealthResponse(response_.endpoint_health_response()
+                                           .cluster_endpoints_health(0)
+                                           .locality_endpoints_health(1)
+                                           .endpoints_health(0),
                                        envoy::config::core::v3::HEALTHY,
                                        host2_upstream_->localAddress()))) {
     ASSERT_TRUE(hds_stream_->waitForGrpcMessage(*dispatcher_, response_));
   }
 
+  // Endpoint health response nested layout assertions.
+  ASSERT_EQ(response_.endpoint_health_response().cluster_endpoints_health_size(), 1);
+  ASSERT_EQ(response_.endpoint_health_response().cluster_endpoints_health(0).cluster_name(),
+            "anna");
+  ASSERT_EQ(response_.endpoint_health_response()
+                .cluster_endpoints_health(0)
+                .locality_endpoints_health_size(),
+            2);
+  ASSERT_EQ(response_.endpoint_health_response()
+                .cluster_endpoints_health(0)
+                .locality_endpoints_health(0)
+                .locality()
+                .sub_zone(),
+            "hobbiton");
+  ASSERT_EQ(response_.endpoint_health_response()
+                .cluster_endpoints_health(0)
+                .locality_endpoints_health(0)
+                .endpoints_health_size(),
+            1);
+  ASSERT_EQ(response_.endpoint_health_response()
+                .cluster_endpoints_health(0)
+                .locality_endpoints_health(1)
+                .locality()
+                .sub_zone(),
+            "emplisi");
+  ASSERT_EQ(response_.endpoint_health_response()
+                .cluster_endpoints_health(0)
+                .locality_endpoints_health(1)
+                .endpoints_health_size(),
```

Comment:
You may avoid these redundancies by using temporary variables, like cluster_endpoints_health, locality_endpoints_health and endpoint_health...

Reply:
What about moving these checks into checkEndpointHealthResponse()?

Reply:
I went ahead and used checkEndpointHealthResponse, as well as pulling in the temp variables to make the intg tests more readable.